include/mega/http.h

```diff
@@ -104,18 +104,28 @@ class MEGA_API SpeedController
     m_off_t getMeanSpeed();
 
     // interval to calculate the mean speed (ds)
-    static const int SPEED_MEAN_INTERVAL_DS;
+    static const int SPEED_MEAN_MAX_INTERVAL_DS = 50;
 
-    // max values to calculate the mean speed
-    static const int SPEED_MAX_VALUES;
+    void requestStarted();
+    m_off_t requestProgressed(m_off_t newPos);
+    m_off_t lastRequestSpeed();
+    dstime requestElapsedDs();
 
 protected:
-    map<dstime, m_off_t> transferBytes;
-    m_off_t partialBytes;
-
-    m_off_t meanSpeed;
-    dstime lastUpdate;
-    int speedCounter;
+    // a circular buffer of bytes received/transmitted per decisecond
+    std::array<m_off_t, SPEED_MEAN_MAX_INTERVAL_DS> circularBuf;
```

Comment:
Please, remember to name variable members as `mCamelCase`. I'll refactorize these ones ;)